src/t_stream.c

```diff
@@ -946,9 +968,23 @@ static int streamParseAddOrTrimArgsOrReply(client *c, streamAddTrimArgs *args, i
         } else if (xadd && !strcasecmp(opt,"nomkstream")) {
             args->no_mkstream = 1;
         } else if (xadd) {
+            /* At this point, we check if the ID is in the form of "ms-*". */
```

Comment:
Didn't read the whole PR yet but I feel this whole block belongs inside streamParseStrictIDOrReply(c,c->argv[i],&args->id,0) (maybe that'll eliminate that dirty hack as well)

Reply:
I was also considering this, but was afraid of the risk/complexity b/c the parser is used all over the place besides XADD.

That said, iiuc, what you mean is something like this sig:

```
int streamGenericParseIDOrReply(client *c, const robj *o, streamID *id, uint64_t missing_seq, int strict, int *seq_given)
```

And if 'seq_given != NULL` then allow the special use case and set it to 0/1. Correct?

Reply:
Resolved via b7277aae2